server/rpc_codec.go

```diff
@@ -96,7 +97,17 @@ func (c *rpcPlusCodec) WriteResponse(r *response, body interface{}, last bool) e
 		Header: map[string]string{},
 	}
 	if err := c.codec.Write(m, body); err != nil {
-		return err
+		c.buf.wbuf.Reset()
+		m = &codec.Message{
```

Comment:
Considering codec.Message is defined just a few lines earlier and a coded should not modify the message in the Write method could you just do `m.Error = errors.Wrapf(err, "Unable to encode body").Error()`.